internal/services/cdn/cdn_frontdoor_security_policy_resource.go

```diff
@@ -232,13 +232,22 @@ func resourceCdnFrontdoorSecurityPolicyRead(d *pluginsdk.ResourceData, meta inte
 
 		wafPolicyId := ""
 		if waf.WafPolicy != nil && waf.WafPolicy.ID != nil {
-			wafPolicyId = *waf.WafPolicy.ID
+			parsedId, err := parse.FrontDoorFirewallPolicyIDInsensitively(*waf.WafPolicy.ID)
+			if err != nil {
+				return fmt.Errorf("flattening `cdn_frontdoor_firewall_policy_id`: %+v", err)
+			}
+			wafPolicyId = parsedId.ID()
 		}
 
 		if waf.Associations != nil {
 			for _, item := range *waf.Associations {
+				domain, err := cdnfrontdoorsecurityparams.FlattenSecurityPoliciesActivatedResourceReference(item.Domains)
+				if err != nil {
+					return fmt.Errorf("flattening `domain`: %+v", err)
```

Comment:
Does this error message seem a little misleading since this can be a `domain` or an `endpoint`?
```suggestion
					return fmt.Errorf("flattening `ActivatedResourceReference`: %+v", err)
```

Reply:
sure, have updated the error message